chain-vote/src/decr_nizk.rs

```diff
@@ -0,0 +1,146 @@
+//! Non-interactive Zero Knowledge proof for correct ElGamal
+//! decryption. We use the notation and scheme presented in
+//! Figure 5 of the Treasury voting protocol spec.
+//!
+//! The proof is the following:
+//!
+//! `NIZK{(pk, C, M), (sk): M = Dec_sk(C) AND pk = g^sk}`
+//!
+//! which makes the statement, the public key, `pk`, the ciphertext
+//! `(e1, e2)`, and the message, `m`. The witness, on the other hand
+//! is the secret key, `sk`.
+#![allow(clippy::many_single_char_names)]
+use super::encryption::Ciphertext;
+use super::gang::{GroupElement, Scalar};
+use crate::encryption::{PublicKey, SecretKey};
+use cryptoxide::digest::Digest;
+use cryptoxide::sha2::Sha512;
+use rand::{CryptoRng, RngCore};
+
+/// Proof of correct decryption.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ProofDecrypt {
```

Comment:
I was wondering if we should have a common interface for ZK proofs, maybe it's too much overhead. A first step could be grouping them under a common directory / module

Reply:
Yes, I agree. Maybe we can do that refactor in a different PR?